File: 2015/05/01.py
with open("input.txt", "r") as file: lines = file.read().strip().splitlines()

total = sum(
    1 for string in lines
    if sum(c in "aeiou" for c in string) >= 3
    and any(string[i] == string[i+1] for i in range(len(string) - 1))
    and all(f not in string for f in ["ab", "cd", "pq", "xy"])
)

print("Part 01: ", total)

File: 2025/02/01.lua
local file = io.open("input.txt", "r")
local total = 0

for range in file:lines() do
    for r in range:gmatch("[^,%s]+") do
        local min, max = r:match("(%d+)-(%d+)")
        for id = tonumber(min), tonumber(max) do
            local s = tostring(id)
            if #s % 2 == 0 then 
                local f_half, l_half = s:sub(1, #s/2), s:sub(#s/2 + 1)
                if f_half == l_half then total = total + id end
            end
        end
    end
end

print("Part 01 : " .. total)